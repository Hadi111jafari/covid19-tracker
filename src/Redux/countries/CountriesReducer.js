import { fetchAllCases } from '../../APIs/CovidAPI';
import { uptoNow } from '../../Logic/Date';
import getMapUrl from '../../APIs/MapsAPI';

const GET_COUNTRIES = 'Covid19_Tracker/countries/GET_COUNTRIES';

let Loading = false;

const sendCountries = (countries) => ({
  type: GET_COUNTRIES,
  payload: countries,
});

const getCountries = () => async (dispatch) => {
  if (Loading) return;
  const data = await fetchAllCases();
  const countriesList = data.dates[uptoNow].countries;
  const allCountries = data.total;
  const mappedAllCountries = {
    id: allCountries.name,
    name: 'All Countries',
    image: getMapUrl('world'),
    today_confirmed: allCountries.today_confirmed.toLocaleString(),
  };
  const countriesMappedList = Object.keys(countriesList).map((key) => ({
    id: key,
    name: countriesList[key].name,
    image: getMapUrl(countriesList[key].name.toLowerCase()),
    today_confirmed: countriesList[key].today_confirmed.toLocaleString(),
  }));
  countriesMappedList.unshift(mappedAllCountries);
  dispatch(sendCountries(countriesMappedList));
  Loading = true;
};

const covidCountriesReducer = (state = [], action) => {
  switch (action.type) {
    case GET_COUNTRIES:
      return action.payload;

    default:
      return state;
  }
};

export { getCountries, GET_COUNTRIES, sendCountries };
export default covidCountriesReducer;
