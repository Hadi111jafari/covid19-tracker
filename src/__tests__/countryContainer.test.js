import renderer from 'react-test-renderer';
import appComponentMock from '../__mocks__/index';
import * as Redux from '../Redux/countries/CountriesReducer';
import '@testing-library/jest-dom';
import countries from '../__mocks__/countries.mock';

jest.spyOn(Redux, 'getCountries')
  .mockImplementation(() => (Redux.dispatchCountries(countries)));

describe('Render App', () => {
  test('Assert snapshot', () => {
    const appSnap = renderer.create(
      <appComponentMock />,
    ).toJSON();
    expect(appSnap).toMatchSnapshot();
  });
});
