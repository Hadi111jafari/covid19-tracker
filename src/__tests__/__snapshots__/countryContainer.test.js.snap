// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Assert snapshot 1`] = `<appComponentMock />`;
